### PR TITLE
Packer console

### LINKED ISSUE #7009: Packer Console
### Feature Request

It would be nice to check the variables or interpolations in the console, similar to `terraform console`. For example, right now I have to run `packer build` and check the variable interpolated values within console output during EBS volume tagging.

Approximate interface:

```
> git-branch
staging

OR

> user.git-branch
staging

OR

> user `git-branch`
staging
```
```
> "{{ user `application` }}-{{timestamp }}"
splash-1542381038
```

### PR DESCRIPTION
**Pull request by SwampDragons**
Implements a `packer console` analogous to the `terraform console` command

Closes #7009

### CODE REVIEW

File: helper/wrappedreadline/wrappedreadline_unix.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	return int(ws.Col)
+}
+
+func init() {
+	// The standard streams are passed in via extra file descriptors.
+	wrappedStdin = os.NewFile(uintptr(3), "stdin")
+	wrappedStdout = os.NewFile(uintptr(4), "stdout")
+	wrappedStderr = os.NewFile(uintptr(5), "stderr")
```
[the remark this answers is not in the document]
Comment (azr) [reply helper/wrappedreadline/wrappedreadline_unix.go]: This seems to be caused by panicwrap: https://github.com/hashicorp/terraform/commit/fd36b548c50595bed1306cde581be94e3bec5a61

[the remark this answers is not in the document]
Comment (azr) [reply helper/wrappedreadline/wrappedreadline_unix.go]: https://github.com/hashicorp/terraform/blob/73a1564dac0ec60f9aa236c465bd42f2e7399552/helper/wrappedreadline/wrappedreadline.go#L16-L22

File: command/console.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+func (c *ConsoleCommand) Run(args []string) int {
+	flags := c.Meta.FlagSet("console", FlagSetVars)
+	flags.Usage = func() { c.Ui.Say(c.Help()) }
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	var templ *template.Template
+
+	args = flags.Args()
```
[the remark this answers is not in the document]
Comment (SwampDragons) [reply command/console.go]: This seems like a problem that probably occurs for all of our command line arguments?

File: helper/wrappedreadline/wrappedreadline_unix.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (SwampDragons) [reply helper/wrappedreadline/wrappedreadline_unix.go]: Do you need more clarification here, or did you answer your own question?

File: command/console.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (azr) [reply command/console.go]: Actually, yes !

File: helper/wrappedreadline/wrappedreadline_unix.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (azr) [reply helper/wrappedreadline/wrappedreadline_unix.go]: I need more clarification pls 🙂 

The way I understand this is that for terraform: panicwrap catches 0,1 & 2 and redirect them to 3,4,5, if the go program is being panicwrapped 🌯 . ( I could not find the code that does this forwarding/multiplexing )

I'm currently wondering if this is the case for packer as we I've only seen 0,1&2 be used.
I'm also wondering if packer should use these wrappedStds.

[the remark this answers is not in the document]
Comment (azr) [reply helper/wrappedreadline/wrappedreadline_unix.go]: I think this is unused/legacy code and now the multiplexing is done through rpc; so we could just remove those 3 lines.

[the remark this answers is not in the document]
Comment (SwampDragons) [reply helper/wrappedreadline/wrappedreadline_unix.go]: It's definitely not vestigial code. If I remove those 3 lines, the REPL just immediately exits.  The reason I added all this copypasta from terraform is that when I initially implemented the REPL using just the UI, the cli functionality was very janky. There may be a better way to do this, but this stream management I think is to provide what https://github.com/chzyer/readline expects -- and that library gives us a lot of nice REPL functionality like history for free.

[the remark this answers is not in the document]
Comment (azr) [reply helper/wrappedreadline/wrappedreadline_unix.go]: Okay ! thanks !

**Pull request merged**